Sources/xcproj/PBXProjObjects+Helpers.swift

```diff
@@ -99,10 +99,14 @@ public extension PBXProj.Objects {
         sourceTree: PBXSourceTree = .group,
         sourceRoot: Path) throws -> ObjectReference<PBXFileReference> {
 
-        guard filePath.isFile else {
-            throw XCodeProjEditingError.notAFile(path: filePath)
+        guard filePath.exists else {
+            throw XCodeProjEditingError.fileNotExists(path: filePath)
         }
 
+        guard let fileType = PBXFileReference.fileType(path: filePath) else {
```

Comment:
I don't think we should require that it's a file we know about. `PBXFileReference.fileType` isn't an exhaustive list of every possible file, and `explicitFileType` and `lastKnownFileType` are optional properties anyway that get filled in by Xcode when the project is opened.